### PR TITLE
Send Results set to pxp frontend

### PR DESCRIPTION
**Pull request by emmastephenson**
# BACKEND PULL REQUEST

## Related Issue

- Part one of #3756 - the UI still needs to be updated to show multiplex results

## Changes Proposed

- Update the PxpResponse object to send the set of results as well as the existing results column.

## Additional Information

- I verified that this change works with removed and corrected results created before #3853, which was our problem before
- This is slightly different from other attempts to send Results to the pxp frontend in two ways:
      1. The type we're sending is a little different. Rather than sending a TestResult for covid, fluA, and fluB specifically, we're just sending the entire set of results. This brings the pxp response in line with the graphQL response (which is important for the print modal - this is why pxp failed to load during our bug bash.)
      2. We're not sending the entire Result object, as when it's serialized by the pxp response it includes too much information (testOrder ID, testEvent id, etc). Instead, a new `SupportedDiseaseTestResult` is created to hold the key information - the disease and the test result.

## Testing

- Patient experience shows no changes
- Patient experience loads for all kinds of tests (corrected, removed, etc)

## Checklist for Primary Reviewer

- [ ] Any large-scale changes have been deployed to `test`, `dev`, or `pentest` and smoke tested
- [ ] Any content updates (user-facing error messages, etc) have been approved by content team
- [ ] Any changes that might generate questions in the support inbox have been flagged to the support team
- [ ] GraphQL schema changes are backward compatible with older version of the front-end
- [ ] Changes comply with the SimpleReport Style Guide
- [ ] Changes with security implications have been approved by a security engineer (changes to  authentication, encryption, handling of PII, etc.)
- [ ] Any dependencies introduced have been vetted and discussed

### CODE REVIEW

File: backend/src/test/java/gov/cdc/usds/simplereport/api/pxp/PatientExperienceControllerTest.java [hunk 1]
```diff
@@ -167,6 +172,148 @@ void verifyLinkV2ReturnsTestResultInfo() throws Exception {
...
+
+    // THEN
+    String requestId =
+        mockMvc
+            .perform(builder)
+            .andExpect(status().isOk())
+            .andExpect(jsonPath("$.testEventId", is(testEvent.getInternalId().toString())))
+            .andExpect(jsonPath("$.results", Matchers.hasSize(3)))
+            .andExpect(
+                jsonPath("$.results[?(@.disease.name == \"COVID-19\" && @.result == \"POSITIVE\")]")
```
Comment (emmastephenson) [inline backend/src/test/java/gov/cdc/usds/simplereport/api/pxp/PatientExperienceControllerTest.java:215]: The results come back as a Set, so unlike some of these other `jsonPaths` we can't assert on the exact order the results will come back in. This regex allows us to assert that somewhere in the set, we expect a disease with the name COVID-19 to exist, and that at that `result.result`, it should be POSITIVE.

**Comment by emmastephenson** [comment]
Reviewers - this is now ready for review/merge, since the correct/removed results that caused us issues before have now been backfilled. Let me know if you have any questions, I'm always happy to hop on a call and explain!

**Review by emyl3 — APPROVED** [review]
Looks good to me from what I can understand! I was able to confirm that `results` were being passed to the frontend. Thank you, Emma! 😹

**Comment by emmastephenson** [comment]
@nathancrtr, @mehansen, @johanna-skylight, friendly ping for a review on this PR :)

File: backend/src/main/java/gov/cdc/usds/simplereport/api/model/pxp/PxpVerifyResponseV2.java [hunk 2]
```diff
@@ -25,6 +30,12 @@ public PxpVerifyResponseV2(Person person, TestEvent testEvent) {
 
     this.testEventId = testEvent.getInternalId();
     this.result = testEvent.getResult();
+    Set<Result> allResults = testEvent.getResults();
+    results = new HashSet<>();
```
Comment (johanna-skylight) [inline backend/src/main/java/gov/cdc/usds/simplereport/api/model/pxp/PxpVerifyResponseV2.java:34]: For what I understand here we are just adding a new field results to the response that will contain an array instead of just one result (similar to the changes done for the test queue related endpoints). Just want to confirm that we will not run in any weird edge case while retrieving the results of the DB when the test is a correction.

File: backend/src/main/java/gov/cdc/usds/simplereport/db/model/auxiliary/SupportedDiseaseTestResult.java [hunk 3]
```diff
@@ -0,0 +1,12 @@
+package gov.cdc.usds.simplereport.db.model.auxiliary;
+
+import gov.cdc.usds.simplereport.db.model.SupportedDisease;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+
+@AllArgsConstructor
+@Getter
+public class SupportedDiseaseTestResult {
```
Comment (johanna-skylight) [inline backend/src/main/java/gov/cdc/usds/simplereport/db/model/auxiliary/SupportedDiseaseTestResult.java:9]: General question. Why is this class to represent a disease result introduced in this PR?. We are already returning disease results for other endpoints so why is this new one introduced?

File: backend/src/test/java/gov/cdc/usds/simplereport/api/pxp/PatientExperienceControllerTest.java [hunk 4]
```diff
@@ -167,6 +172,148 @@ void verifyLinkV2ReturnsTestResultInfo() throws Exception {
...
+            .characterEncoding("UTF-8")
+            .content(requestBody);
+
+    // THEN
+    String requestId =
+        mockMvc
+            .perform(builder)
+            .andExpect(status().isOk())
+            .andExpect(jsonPath("$.testEventId", is(testEvent.getInternalId().toString())))
+            .andExpect(jsonPath("$.results", Matchers.hasSize(3)))
```
Comment (nathancrtr) [inline backend/src/test/java/gov/cdc/usds/simplereport/api/pxp/PatientExperienceControllerTest.java:213]: Think it's worth running an assertion on `$.result` too?

**Review by nathancrtr — APPROVED** [review]
LGTM! I manually inserted additional `Result`s into my local database to force a multiplex test event and could see both entries in the `results` array returned in the PXP verification response. The top-level `result` correctly reflects the Covid-19 test result, regardless of the ordering of the `results` array. Tests look great too 👏

**Review by emyl3 — APPROVED** [review]
re-approving!

File: backend/src/main/java/gov/cdc/usds/simplereport/api/model/pxp/PxpVerifyResponseV2.java (hunk 2 above)
Comment (emmastephenson) [reply backend/src/main/java/gov/cdc/usds/simplereport/api/model/pxp/PxpVerifyResponseV2.java:34]: Just verified this - for removed events, we're still displaying the patient experience correctly and only returning a single result.
